src/proto/single/mod.rs

```diff
@@ -266,6 +288,17 @@ impl Job {
     pub fn failure(&self) -> &Option<Failure> {
         &self.failure
     }
+
+    /// Error message for this job, if any.
+    ///
+    /// A convenience method that, internally, will check if this
+    /// job has got a [`Failure`] and - if it does - will access
+    /// the failure's message (see [`Failure::message`]).
```

Comment:
I don't know if this convenience method is worthwhile?

Reply:
Maybe we can nuke it, I will then use a helper function in the tests binaries